spec/libraries/acts_as_conditional_spec.rb

```diff
@@ -2,34 +2,41 @@
 require 'rails_helper'
 
 RSpec.describe 'Extension: Acts as Conditional', type: :model do
-  let(:instance) { create(:instance) }
-
-  with_tenant(:instance) do
-    let(:course) { create(:course) }
-    let(:achievement) do
-      achievement = build(:achievement, course: course)
-      create(:achievement_condition, achievement: achievement)
-      achievement
+  class self::DummyConditionalClass < ActiveRecord::Base
+    def self.columns
+      []
     end
 
-    describe '#conditions' do
-      subject { achievement.conditions }
+    acts_as_conditional
+  end
 
-      it 'is of type Condition' do
-        expect(subject).to all be_instance_of(Course::Condition)
-      end
+  class self::DummyConditionClass < ActiveRecord::Base
+    def self.columns
+      []
     end
 
-    describe '#specific_conditions' do
-      subject { achievement.specific_conditions }
+    acts_as_condition
+  end
+
+  describe 'objects which act as conditional' do
+    subject { self.class::DummyConditionalClass.new }
 
+    it { is_expected.to respond_to(:conditions) }
```

Comment:
Do we usually test the behavior for associations? Not sure if there are any good way to do that without using a concrete object.